### ISSUE TITLE
add option to hide status bar

### ISSUE THREAD

**Issue by slymax** [zed-industries/zed#8169]
### Check for existing issues

- [X] Completed

### Describe the feature

there should be an option to hide the status bar

#5120 and #6862 are related but not quite the same

**Comment by jansol** [comment]
#4382 has a list for tracking hide-able UI elements and one of the items on the list is the status bar.

**Comment by mikayla-maki** [comment]
Did a little poking at my very simple VSCode install (only plugins are copilot and copilot chat, which I don't think I ever installed myself, and rust-analyzer) to learn how they handle this:

- They have very little information in the status bar by default. Just LSP stuff, file information (spaces, encoding, etc.), branches, and SSH connections status.  
- Most of these have alternative UI that could fill the same role, e.g. their git panel can be configured to show the repository and branch, the LSP indicator is on the editor rather than the status bar, etc.
- They have a "show notifications" action to pop up what rust analyzer is doing right now
- They have a problems tab in the bottom panel that can show diagnostics info
- They also moved their copilot settings out of the status bar and into a menu
- Interestingly, there's no UI that covers all of the info that's available in the status bar, but there is UI available for configuring the various options, e.g. file encoding.

By contrast, Zed puts a lot more in the status bar right now:

- Update available messages
- Language server statuses and errors with no "show notifications" equivalent
- Diagnostics (there's the tab, but there's no quick way to reference them)
- Edit Prediction and copilot setting controls

And none of these have alternative UIs. Fixing this bug will require addressing these (or consciously choosing not to address them) in some way

**Comment by zzzint** [comment]
@mikayla-maki Casting my vote for what's being shown on the status bar being easily changeable. Diagnostics & LSP info in the status bar being constantly updated in the corner of my eye while typing has been the thing holding me back from moving over to Zed for quite a while now.

**Comment by slymax** [comment]
when I originally created the issue, the icons in the status bar where all wonky and misaligned. I really don't mind the current state of the status bar though – but yeah some more options in the config as to what items are visible might be useful

**Comment by ksweetie** [comment]
I would push back on the idea that every feature currently in the status bar needs an alternative UI outside of the status bar in order to add this feature. As long as there's a command to show/hide the bar, we could toggle it on with a shortcut as needed.

99.9% of the time, I don't need anything in the status bar, and like @zzzint said, I find the diagnostics and LSP info distracting. I already don't use any of the buttons in the status bar, because any that I would use I have keyboard shortcuts for.

**Comment by somtooo** [comment]
> I would push back on the idea that every feature currently in the status bar needs an alternative UI outside of the status bar in order to add this feature. As long as there's a command to show/hide the bar, we could toggle it on with a shortcut as needed.
> 
> 99.9% of the time, I don't need anything in the status bar, and like [@zzzint](https://github.com/zzzint) said, I find the diagnostics and LSP info distracting. I already don't use any of the buttons in the status bar, because any that I would use I have keyboard shortcuts for.

yeah trying to understand why a separate place is needed for it. If you hid it thats on you. lol

**Comment by notpeter** [comment]
Closing as duplicate of:
- https://github.com/zed-industries/zed/issues/5120

As it has 260 upvotes and this issue has 22 👍 
Please 👍 and subscribe to that issue to track its progress.
